compiler/luci/service/src/Nodes/CircleReshape.cpp

```diff
@@ -34,4 +40,100 @@ luci::CircleNode *CloneNodeLet<CN::OPQR>::visit(const luci::CircleReshape *node)
   return cloned;
 }
 
+namespace sinf
+{
+
+loco::TensorShape Algorithm::visit(const luci::CircleReshape *node)
+{
+  LOGGER(l);
+
+  const loco::DataType S32 = loco::DataType::S32;
+
+  loco::TensorShape shape_by_input;
+  {
+    LUCI_ASSERT(node->shape(), "2nd input shape() should not be nullptr");
+
+    // Only support node's shape() is CircleConst with S32
+    // TODO support other node with other types
+    auto const_shape_node = dynamic_cast<luci::CircleConst *>(node->shape());
+    if (const_shape_node != nullptr)
+    {
+      LUCI_ASSERT(const_shape_node->dtype() == S32, "Only support int32 CircleConst");
+
+      shape_by_input.rank(const_shape_node->size<S32>());
+
+      for (uint32_t axis = 0; axis < shape_by_input.rank(); ++axis)
+      {
+        shape_by_input.dim(axis) = const_shape_node->at<S32>(axis);
+        if (const_shape_node->at<S32>(axis) < 0)
+          shape_by_input.dim(axis).unset();
+      }
+    }
+    else
+    {
+      // We use shape from the node itself
+      shape_by_input = own_shape(node);
+    }
+  }
+
+  loco::TensorShape shape_by_attr;
+  {
+    shape_by_attr.rank(node->newShape()->rank());
+
+    for (uint32_t axis = 0; axis < shape_by_attr.rank(); ++axis)
+    {
+      shape_by_attr.dim(axis) = node->newShape()->dim(axis);
+    }
+  }
+
+  if (!(shape_by_input == shape_by_attr))
+  {
+    INFO(l) << "CircleReshape: Two new shape information mismatched : " << std::endl;
+    INFO(l) << "   shape_by_input : " << shape_by_input << std::endl;
+    INFO(l) << "   shape_by_attr : " << shape_by_attr << std::endl;
+  }
```

Comment:
I think the else-statement maybe substituted by `shape_by_attr` logic according to https://github.com/tensorflow/tensorflow/blob/2fe29379f6449d263b787eb636020fd81017b8bf/tensorflow/lite/kernels/reshape.cc#L137-L143

```suggestion
  loco::TensorShape output_shape;
  {
    LUCI_ASSERT(node->shape(), "2nd input shape() should not be nullptr");

    // Only support node's shape() is CircleConst with S32
    // https://github.com/tensorflow/tensorflow/blob/2fe29379f6449d263b787eb636020fd81017b8bf/tensorflow/lite/kernels/reshape.cc#L130
    auto const_shape_node = dynamic_cast<luci::CircleConst *>(node->shape());
    if (const_shape_node != nullptr)
    {
      LUCI_ASSERT(const_shape_node->dtype() == S32, "Only support int32 CircleConst");

      output_shape.rank(const_shape_node->size<S32>());

      for (uint32_t axis = 0; axis < output_shape.rank(); ++axis)
      {
        output_shape.dim(axis) = const_shape_node->at<S32>(axis);
        if (const_shape_node->at<S32>(axis) < 0)
          output_shape.dim(axis).unset();
      }
    }
    else
    {
      output_shape.rank(node->newShape()->rank());
      for (uint32_t axis = 0; axis < output_shape.rank(); ++axis)
      {
        output_shape.dim(axis) = node->newShape()->dim(axis);
      }
    }
  }
```